collect_stata/read_stata.py

```diff
@@ -0,0 +1,156 @@
+"""read_stata.py"""
+__author__ = "Marius Pahl"
+
+import logging
+import re
+
+import numpy as np
+import pandas as pd
+
+LOGGER = logging.getLogger(__name__)
+
+
+def cat_values(var, varscale, datatable, data):
+    """
+    Extrat categorical metadata from stata files
+
+    Input:
+    var: string
+    varscale: dict
+    datatable: pandas DataFrame
+    data: pandas StataReader
+
+    Output:
+    cat_list: list
+    """
+
+    cat_list = list()
+
+    df_nomis = datatable[var].copy()
+
+    for index, value in enumerate(df_nomis):
+        if not isinstance(value, str) and value < 0:
+            df_nomis[index] = np.nan
```

Comment:
What is `df_nomis` and what happens to it?

Reply:
fragment of the past...